### PR TITLE
Aerogear 2375 remove need for labelling service instance

### PR DESCRIPTION
**Pull request by maleck13**
**Describe what this PR does and why we need it**:

Changes proposed in this pull request
 - remove the dependency on a labelled service instance

**VERIFICATION**

- [x] point your service broker at maleck13 org, `oc edit configmap broker-config -n ansible-service-broker`
- [x] deploy keycloak and fh-sync
- [x] check the service instance are not labelled with the serviceName label
- [x] run `make build_binary` in this repo
- [x] run `mobile --namespace=something get serviceinstances keycloak `
- [x] run `mobile --namespace=something get serviceinstances fh-sync-server `
- [x] run `mobile --namespace=something create integration fh-sync-id keycloak-id --auto-redeploy=true `
- [x] ensure the binding was created
- [x] ensure sync was restarted and says connecting to keycloak in logs
- [x] run `mobile --namespace=something delete integration fh-sync-id keycloak-id --auto-redeploy=true`
- [x] ensure binding is deleted
- [x] ensure pod preset deleted
- [x] ensure sync restarted with not keycloak details

### CODE REVIEW

File: pkg/cmd/integration.go [hunk 1]
```diff
@@ -102,6 +101,27 @@ func podPreset(objectName, secretName, providerSvcName, consumerSvcName string)
 	return &podPreset
 }
 
+func (bc *IntegrationCmd) getServiceNameFromServiceInst(si *v1beta1.ServiceInstance) (string, error) {
+	serviceClass, err := bc.
+		scClient.
```
Comment (vchepeli) [inline pkg/cmd/integration.go]: not better use dots(.) at beginning?
```go
bc.scClient
.ServicecatalogV1beta1()
.ClusterServiceClasses()
.Get(si.Spec.ClusterServiceClassRef.Name, metav1.GetOptions{})
```

File: pkg/cmd/integration.go [hunk 2]
```diff
@@ -102,6 +101,27 @@ func podPreset(objectName, secretName, providerSvcName, consumerSvcName string)
...
+		ServicecatalogV1beta1().
+		ClusterServiceClasses().
+		Get(si.Spec.ClusterServiceClassRef.Name, metav1.GetOptions{})
+	if err != nil {
+		return "", errors.WithStack(err)
+	}
+	serviceMeta := map[string]interface{}{}
+	if err := json.Unmarshal(serviceClass.
+		Spec.
```
Comment (vchepeli) [inline pkg/cmd/integration.go]: same here, but not sure about right format

File: pkg/cmd/integration_test.go [hunk 3]
```diff
@@ -32,6 +33,8 @@ func TestIntegrationCmd_CreateIntegrationCmd(t *testing.T) {
 		},
 	}
 
+	fmt.Print(defaultServiceBinding)
```
Comment (vchepeli) [inline pkg/cmd/integration_test.go]: do we need his in test?

File: pkg/cmd/services.go [hunk 4]
```diff
@@ -443,6 +443,20 @@ Run the "mobile get serviceinstances" command from this tool to see which servic
 	}
 }
 
+func findServiceInstanceByExternalName(client versioned.Interface, ns, name string) ([]v1beta1.ServiceInstance, error) {
+	sis, err := client.ServicecatalogV1beta1().ServiceInstances(ns).List(metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	var relevantSi []v1beta1.ServiceInstance
+	for _, s := range sis.Items {
```
Comment (vchepeli) [inline pkg/cmd/services.go]: Probably better extract _Filter_ function?
```go
func Filter(svcs []v1beta1.ServiceInstance, f func(v1beta1.ServiceInstance) bool) []v1beta1.ServiceInstance {
    rel := make([]v1beta1.ServiceInstance, 0)
    for _, s := range svcs {
        if f(s) {
            rel = append(rel, s)
        }
    }
    return rel
}
```
Then use it like
```go
Filter(sis.Items, func(svc v1beta1.ServiceInstance) bool {
        return svc.Spec.ClusterServiceClassExternalName == SOME_VALUE
})
```

File: pkg/cmd/integration.go (hunk 1 above)
Comment (maleck13) [reply pkg/cmd/integration.go]: cant do that with go

File: pkg/cmd/integration_test.go (hunk 3 above)
Comment (maleck13) [reply pkg/cmd/integration_test.go]: will remove

File: pkg/cmd/services.go (hunk 4 above)
Comment (maleck13) [reply pkg/cmd/services.go]: I think it is ok this way. Not sure we gain too much from the filter function

**Pull request merged**